Allow decoration of functions beyond the admin login

MattBlack85 opened the pull request:
@kencochrane as discussed in #85

kencochrane commented on defender/tests.py:
```diff
@@ -724,6 +729,81 @@ def test_disable_username_lockout(self):
 
+    @patch('defender.config.BEHIND_REVERSE_PROXY', True)
+    @patch('defender.config.FAILURE_LIMIT', 3)
+    def test_login_blocked_for_non_standard_login_views_without_msg(self):
```
Can you add docstrings to the test methods explaining what each test does? Not all tests have them now, and I have been meaning to fix that.

MattBlack85 replied:
Sure, no problem

kencochrane replied:
Thx

kencochrane commented on defender/middleware.py:
```diff
@@ -18,9 +17,10 @@ def __init__(self, *args, **kwargs):
                 LoginView.dispatch = watch_login_method(LoginView.dispatch)
             except ImportError:  # Django < 1.11
-                auth_views.login = watch_login(auth_views.login)
+                auth_views.login = watch_login()(auth_views.login)
```
Is that correct? I'm no expert when it comes to decorators so might be correct, it just looks weird with `watch_login()(auth_views.login)`

MattBlack85 replied:
That looks weird but it's correct. The old decorator couldn't accept arguments and had a deep of 2. I had to add one more level in order to accept arguments so the first thing the new decorator expects now are the arguments (or the defaults) while here a function was passed in first instance. Does it make sense?

kencochrane replied:
Yes, thank you.

kencochrane commented:
It looks like coverage dropped a little, and a few more quantified code minor issues were found. Other than that, pretty good, thanks for the PR.

kencochrane commented:
I just merged another PR, it looks like there is a conflict.

MattBlack85 commented:
@kencochrane please review again, I added the docstrings but also 2 more commits.
One is excluding test files from coverage which is dropping coverage by 5% but it gives real results now and the latest one is something I discover yesterday while I was testing my PR, we weren't testing against django 1.11, never, becasue we call `python setup.py develop` which was installing `django>=1.8, django<=1.10`; fixed that

kencochrane commented:
PR looks good, I'll leave it open for a bit in case anyone else wants to review. If no one else comments, I'll merge.

MattBlack85 commented:
:+1:

mehtamanan commented:
Can I use this yet?
